### ISSUE TITLE
Uncaught TypeError: Failed to execute 'uniform4fv' on 'WebGL2RenderingContext': Overload resolution failed.

### ISSUE THREAD

**Issue by Ono-Sendai** [emscripten-core/emscripten#21567]
Hi,
I updated EmSDK to the latest version (3.1.56), and now am hitting the following error while executing my WASM program:

`
gui_client.js:11877 Uncaught TypeError: Failed to execute 'uniform4fv' on 'WebGL2RenderingContext': Overload resolution failed.
    at _glUniform4fv (gui_client.js:11877:8)
`

EmSDK version:

Resolving SDK version '3.1.56' to 'sdk-releases-9d106be887796484c4aaffc9dc45f48a8810f336-64bit'
Installing SDK 'sdk-releases-9d106be887796484c4aaffc9dc45f48a8810f336-64bit'..

It worked fine before updating.

Cheers,
  Nick

**Comment by Ono-Sendai** [comment]
Seems to happen when the _count_ argument to glUniform4fv is 0.

**Comment by sbc100** [comment]
Could you share the full set of link flags you are using?

Can you confirm that this doesn't happen in 3.1.55?  Perhaps you could help track down the exact change that caused this failure by bisecting: https://emscripten.org/docs/contributing/developers_guide.html#bisecting

Finally perhaps you could shared what is on line 11877 of your JS file?  How many arguments are being passed to `uniform4fv` and maybe you could `console.error(arguments)` and report the types you see?

**Comment by Ono-Sendai** [comment]
Here's the JS file function:

```
/** @suppress {duplicate } */ function _glUniform4fv(location, count, value) {
 value >>>= 0;
 if (count <= 72) {
  var view = miniTempWebGLFloatBuffers[4 * count - 1];
  var heap = HEAPF32;
  value = ((value) >>> 2);
  for (var i = 0; i < 4 * count; i += 4) {
   var dst = value + i;
   view[i] = heap[dst >>> 0];
   view[i + 1] = heap[dst + 1 >>> 0];
   view[i + 2] = heap[dst + 2 >>> 0];
   view[i + 3] = heap[dst + 3 >>> 0];
  }
 } else {
  var view = HEAPF32.subarray((((value) >>> 2)) >>> 0, ((value + count * 16) >>> 2) >>> 0);
 }
 GLctx.uniform4fv(webglGetUniformLocation(location), view);
}
```

link flags:

```
[1/1] cmd.exe /C "cd . && C:\programming\emsdk\upstream\emscripten\em++.bat -I"c:/programming/libjpeg-turbo/libjpeg-turbo-emscripten-build/include" -I"c:/code/glare-core/libpng" -I"c:/code/glare-core/pugixml/src" -I"c:/code/glare-core/zlib" -I"c:/code/glare-core/Imath/src/Imath" -I"c:/code/glare-core/OpenEXR/src/lib/Iex" -I"c:/code/glare-core/OpenEXR/src/lib/IlmThread" -I"c:/code/glare-core/OpenEXR/src/lib/OpenEXR" -I"c:/code/glare-core/OpenEXR/src/lib/OpenEXRUtil" -I"c:/code/glare-core/Imath/config_linux" -I"c:/code/glare-core/OpenEXR/config_linux" -I"c:/code/glare-core/" -I"c:/code/glare-core/utils" -I"c:/code/glare-core/networking" -I"c:/code/glare-core/maths" -I"c:/code/glare-core/opengl" -I"c:/code/glare-core/giflib/lib" -I"c:/code/glare-core/little_cms/include" -I"c:/code/glare-core/zstd/lib" -I"c:/code/glare-core/zstd/lib/common" -I"c:/code/glare-core/opencl/khronos" -I"c:/programming/BugSplat/BugSplat/inc" -Ic:/programming/llvm/llvm_15_0_7_dylib_install/include -I/usr/local/include -Ic:/programming/LibreSSL/libressl-3.5.2-install/include -O2 -g -DNDEBUG   -O2 -g -DNDEBUG -DBUILD_TESTS=1 "-s USE_WEBGL=1" -sMIN_WEBGL_VERSION=2 -sMAX_WEBGL_VERSION=2 -sGL_ENABLE_GET_PROC_ADDRESS -fwasm-exceptions -sINITIAL_MEMORY=3500MB -pthread -sPTHREAD_POOL_SIZE=26 -sSTACK_SIZE=262144 -lwebsocket.js --preload-file data @CMakeFiles\gui_client.rsp -o C:\programming\cyberspace\output\test_builds\gui_client.js  && cd ."
```

**Comment by Ono-Sendai** [comment]
printing out stuff to the console, it looks like 'view' is undefined just before the  GLctx.uniform4fv call.
I guess this is due to  ` var view = miniTempWebGLFloatBuffers[4 * count - 1];` when count is zero.

**Comment by Ono-Sendai** [comment]
Am not going to do any version bisection sorry :)

**Comment by sbc100** [comment]
Thank you for the information.  I think that should provide enough info to debug further.

**Comment by sbc100** [comment]
It looks like this was originally fixed in #16837, but the fix didn't cover the path when WEBGL_USE_GARBAGE_FREE_APIS was not available.

**Comment by sbc100** [comment]
I tracked it down.

This was originally reported in #16799, partially fixed in #16837, then exposed again with #21445 (although it continued to exist for webgl1 users).  Latest fix is now in #21573

**Comment by Ono-Sendai** [comment]
great :)
